x/mongo/driver/topology/topology.go

```diff
@@ -169,23 +169,56 @@ func (t *Topology) Connect() error {
 		t.fsm.Servers = append(t.fsm.Servers, description.NewDefaultServer(addr))
 	}
 
-	// store new description
-	newDesc := description.Topology{
-		Kind:                  t.fsm.Kind,
-		Servers:               t.fsm.Servers,
-		SessionTimeoutMinutes: t.fsm.SessionTimeoutMinutes,
-	}
-	t.desc.Store(newDesc)
-	t.publishTopologyDescriptionChangedEvent(description.Topology{}, t.fsm.Topology)
-	for _, a := range t.cfg.seedList {
-		addr := address.Address(a).Canonicalize()
-		err = t.addServer(addr)
-		if err != nil {
+	switch {
+	case t.cfg.loadBalanced:
+		// In LoadBalanced mode, we mock a series of events: TopologyDescriptionChanged from Unknown to LoadBalanced,
```

Comment:
Clever!

Reply:
Very cool. I see no reference to this mocking of initial state changes in the technical design; am I missing it? Or, is this something that's specific to Go?